File: RepoTransBench/source_projects/Python/rust-lang_homu/public_tests/test_public_auth.py
import pytest

from homu import auth as auth_mod

def test_secret_hash_and_check_public():
    secret = "another_secret_string"
    encoded = auth_mod.secret_hash(secret)
    assert isinstance(encoded, str)
    assert encoded != "" and secret not in encoded
    assert auth_mod.check_encoded_secret(secret, encoded)
    assert not auth_mod.check_encoded_secret("wrong_public_secret", encoded)

File: RepoTransBench/target_projects/Python/Rust/rust-lang_homu/public_tests/test_public_auth.rs
use homu_rust::auth;

#[test]
fn test_secret_hash_and_check_public() {
    let secret = "another_secret_string";
    let encoded = auth::secret_hash(secret);
    assert!(!encoded.is_empty());
    assert!(!encoded.contains(secret));
    assert!(auth::check_encoded_secret(secret, &encoded));
    assert!(!auth::check_encoded_secret("wrong_public_secret", &encoded));
}